### PR TITLE
enable heat service based on new systemd_service module

### PR DESCRIPTION
**Pull request by panxia6679**

### CODE REVIEW

File: roles/heat/tasks/main.yml [hunk 1]
```diff
@@ -13,47 +13,52 @@
...
-    port: "{{ item }}/tcp"
-  with_items:
-    - "{{ endpoints.heat.port.haproxy_api }}"
-    - "{{ endpoints.heat_cfn.port.backend_api }}"
-  tags: firewall
-  when: os == 'rhel'
+- block:
+  - name: permit access to heat
+    ufw: rule=allow to_port={{ item }} proto=tcp
+    tags: ufw
```
Comment (nirajdp76) [inline roles/heat/tasks/main.yml]: change this to tags: firewall

**Comment by nirajdp76** [comment]
ok to test

File: roles/heat/tasks/main.yml [hunk 2]
```diff
@@ -13,47 +13,52 @@
...
+      name: "{{ item.name }}"
+      description: "{{ item.desc }}"
+      type: "{{ item.type }}"
+      user: "{{ item.user }}"
+      cmd: "{{ item.cmd }}"
+      config_dirs: "{{ item.config_dirs }}"
+      config_files: "{{ item.config_files }}"
+      restart: "{{ item.restart }}"
+    with_items:
+      - "{{ heat.services.heat_api }}"
```
Comment (nirajdp76) [inline roles/heat/tasks/main.yml]: There are other places in this task where we need to remove os specific tasks of stopping and starting services and replace with os neutron task feeding in service variable for service name. See nova/neutron tasks/main.yml for reference.  U can search for openstack-heat-api in here.

Also, for start heat services add enabled: True to get service enabled to auto start on reboot.

Comment (panxia6679) [inline roles/heat/tasks/main.yml]: thanks for your comments. I forgot to refine start/stop services with new variables. Updated them and please review again.

**Comment by panxia6679** [comment]
looks there is something wrong with CI. it failed when creating stack for testing the PR.

can we have a retest? Thanks.

**Comment by nirajdp76** [comment]
retest

**Pull request merged**